popup.js

```diff
@@ -117,12 +117,13 @@ function adddate(){
     var m=months[dat.getMonth()];
     var hr=dat.getHours();
     var min=dat.getMinutes();
+    var pr= hr-12;
     if(d<10){d="0"+d;}
     if(min<10){min="0"+min;}
-    if (hr>12){hr=hr-12;}
+    if (hr>12){hr="0"+pr;}
     else if (hr==12){hr=hr+":"+min+" PM";}
     else {hr=hr+":"+min+" AM";}
-    if(hr<12){hr= "0"+hr+":"+min+" PM";}
+    if(hr<12){hr= hr+":"+min+" PM";}
```

Comment:
Your changes are still messed up. For AM time you are not adding zero to hr. For hr > 12 you are not adding min and PM to it.

Take a new variable "time" and assign time to it instead of hr variable which has been done originally. 
Make three if statements for time (hr == 12), (hr < 12) and (hr > 12). Then do proper operations with hr and then assign proper time to "time" variable. Like when hr is 8. Do it under (hr < 12)
hr = "0"+hr;
time = hr+":"+min+" AM"